### PR TITLE
Fix: Duplicated Android Breadcrumbs with no Mechanism

### LINKED ISSUE #952: Breadcrumbs are duplicated if there is no mechaism on Android
A possible fix is here https://github.com/getsentry/sentry-dart/pull/936 but this is causing a [flaky CI](https://github.com/getsentry/sentry-dart/pull/938) and we have to figure out what's going on.
I'm [reverting](https://github.com/getsentry/sentry-dart/pull/940) the PR to unblock new releases.

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Closes #952

## :green_heart: How did you test it?

Added unit tests.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by denrase** [comment]
@marandaneto The "CI" issue was that we would run on "Android" as a platform, and we changed the behaviour to removing breadcrumbs if there are no mechanisms, which is also true if there are no exceptions. I have set a fixed platform (iOS) now for this particular unit test file.

File: dart/lib/src/sentry_client.dart [hunk 1]
```diff
@@ -378,12 +378,13 @@ class SentryClient {
...
   SentryEvent _eventWithRemovedBreadcrumbsIfHandled(SentryEvent event) {
-    final exceptions = event.exceptions ?? [];
-    final handled = exceptions.isNotEmpty
-        ? exceptions.first.mechanism?.handled == true
-        : false;
+    final mechanisms =
+        (event.exceptions ?? []).map((e) => e.mechanism).whereType<Mechanism>();
+    final hasNoMechanism = mechanisms.isEmpty;
+    final hasOnlyHandledMechanism =
+        mechanisms.every((e) => (e.handled ?? false));
```
Comment (marandaneto) [inline dart/lib/src/sentry_client.dart]: The default should still be `true` here otherwise the same bug would happen.

Comment (denrase) [reply dart/lib/src/sentry_client.dart]: Is a mechanism that has no `handled` flag (null) considered as handled?

Comment (marandaneto) [reply dart/lib/src/sentry_client.dart]: yes, it's only not handled (app crashed) if `handled = false`, everything else is considered that the error was handled (the app didn't crash).

**Review by marandaneto — APPROVED** [review]

**Pull request merged**